lib/std/debug.zig

```diff
@@ -1677,6 +1692,17 @@ pub const ModuleDebugInfo = switch (builtin.os.tag) {
             unreachable;
         }
     },
+    .haiku => struct {
+        // Haiku should implement dl_iterat_phdr (https://dev.haiku-os.org/ticket/15743)
```

Comment:
Not really, see above.

Reply:
for the time being, i have backed most of the changes except for more minimal stub logic for debug and will try to tackle this as a separate item (per above)